### PR TITLE
DOC: add detail for 'where' argument in ufunc

### PR DESCRIPTION
**Pull request by ThibHlln**
Proposal to replace "leave the value in the output alone" by "leave the value unchanged in the output", and to add more detail about the origin of the unchanged value (especially indicating that if no `out` location is given, the array freshly-allocated by the ufunc is uninitialized).

This commit is related to ticket #12022.

### CODE REVIEW

**Comment by charris** [comment]
LGTM, maybe add that `where` is broadcast over the input array.

**Comment by mattip** [comment]
synchronized, mentioned broadcast in both places, rebased and squashed.

**Comment by charris** [comment]
@mattip What happened to "leaves the values unchanged"? I think  "unchanged" is a better word than "alone", but it will take a bit of a rewrite to make it fit in.

**Comment by mattip** [comment]
changed the phrasing

File: numpy/core/_add_newdocs.py [hunk 1]
```diff
@@ -5655,10 +5655,10 @@ def luf(lamdaexpr, *args, **kwargs):
...
         allocated by the ufunc.
     where : array_like, optional
-        Values of True indicate to calculate the ufunc at that position, values
-        of False indicate to leave the value in the output alone.  Note that if
-        an uninitialized return array is created via the default ``out=None``,
-        then the elements where the values are False will remain uninitialized.
+        Broadcast over the input. If the value is True, output will be set
+        to the ufunc output; otherwise it is left unchanged. Note that if an
+        uninitialized return array is created via the default ``out=None``,
+        output where the values are False will remain uninitialized.
```
[the remark this answers is not in the document]
Comment (mattip) [reply numpy/core/_add_newdocs.py]: changed

File: numpy/core/_add_newdocs.py [hunk 2]
```diff
@@ -5655,10 +5655,11 @@ def luf(lamdaexpr, *args, **kwargs):
...
     where : array_like, optional
-        Values of True indicate to calculate the ufunc at that position, values
-        of False indicate to leave the value in the output alone.  Note that if
-        an uninitialized return array is created via the default ``out=None``,
-        then the elements where the values are False will remain uninitialized.
+        The condition is broadcast over the input. If the value is True, output
+        will be set to the ufunc output; otherwise it is left unchanged. Note
+        that if an uninitialized return array is created via the default
+        ``out=None``, output at the position where the values are False will
+        remain uninitialized.
```
Comment (eric-wieser) [inline numpy/core/_add_newdocs.py]: Better. How about:
```
This condition is broadcast over the input. At locations where the
condition is True, the `out` array will be set to the ufunc result.
Elsewhere, the `out` array will retain its original value.
Note that if an uninitialized `out` array is created via the default
``out=None``, locations within it where the condition is False will
remain uninitialized.
```

With the changes being:

* "condition" helps separate it from the normal "value" arguments, `x1..xn`
* pluralizing things makes this a little clearer.
* Referring to the `out` array explicitly maybe helps emphasize where the uninitialized result comes from.

Perhaps this is just bike-shedding though.

Comment (mattip) [reply numpy/core/_add_newdocs.py]: adopted

Comment (mattip) [reply numpy/core/_add_newdocs.py]: .. and synced the other `where` docstring as well

**Comment by mattip** [comment]
Thanks @ThibHlln

**Pull request merged**